src/hobbits-plugins/analyzers/WidthFramer/widthframerform.cpp

```diff
@@ -141,46 +140,75 @@ void WidthFramerForm::widthSelected(QModelIndex index)
 QVector<QPointF> WidthFramerForm::autocorrelate(QSharedPointer<const BitArray> bits)
 {
     int N = 1 << 19;
-    fftw_complex *fft_in = reinterpret_cast<fftw_complex*>(fftw_malloc(sizeof(fftw_complex) * unsigned(N)));
-    fftw_complex *fft_out = reinterpret_cast<fftw_complex*>(fftw_malloc(sizeof(fftw_complex) * unsigned(N)));
 
-    fftw_plan fft_plan1 = fftw_plan_dft_1d(N, fft_in, fft_out, FFTW_FORWARD, FFTW_ESTIMATE);
-    fftw_plan fft_plan2 = fftw_plan_dft_1d(N, fft_in, fft_out, FFTW_BACKWARD, FFTW_ESTIMATE);
+    //create set up for PFFFT
+    PFFFT_Setup *setup = pffft_new_setup(N, PFFFT_COMPLEX);
+    
+    if(!setup){
+        //return an empty vector
+        QVector<QPointF> badSetup(N / 2);
+        return badSetup;
+    }
 
-    // prepare and run first FFT
-    for (int i = 0; i < N; i++) {
-        fft_in[i][0] = 0;
-        fft_in[i][1] = 0;
+    //allocate the arrays, or "float buffers," for input, output, and work
+    float *input = (float*)pffft_aligned_malloc(N * 2 * sizeof(float));
+    float *output = (float*)pffft_aligned_malloc(N * 2 * sizeof(float));
+    float *work= (float*)pffft_aligned_malloc(N * 2 * sizeof(float));
+
+    //if any float buffers are null return an empty vector
+    if(!input){
+        QVector<QPointF> nullInput(N / 2);
+        return nullInput;
+    }
+    if(!output){
```

Comment:
you might also want to combine these like `!input || !output || !work` since it is all sort of the same check/result.